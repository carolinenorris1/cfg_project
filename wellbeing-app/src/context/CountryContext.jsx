import { createContext, useState } from "react";

const CountryContext = createContext();

    export function CountryProvider({ children }) {
        const [country, setCountry] = useState("UK");
        return (
            <CountryContext.Provider value={ { country, setCountry } }>{children}</CountryContext.Provider>
        );
    }   

export default CountryContext;