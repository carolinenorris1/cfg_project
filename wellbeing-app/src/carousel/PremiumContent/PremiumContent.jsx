import React, { useContext } from "react";
import CountryContext from "../../context/CountryContext";

function PremiumContent() {
  const context = useContext(CountryContext);
  const country = context.country;

  function getPremiumPrice(country) {
    let premiumPrice;

    if (country === "UK") {
      premiumPrice = "£2.99/month";
    } else if (country === "US") {
      premiumPrice = "$4.99/month";
    } else if (country === "EU") {
      premiumPrice = "€3.99/month";
    } else {
      premiumPrice = "*no region selected*";
    }
    return premiumPrice;
  }
  return (
    <div>
      <p>Want to discover even more great features?</p>
      <p>
        For just <b>{getPremiumPrice(country)}</b>, you can unlock even more
        tools to help you <b>UnHook</b>.
      </p>
    </div>
  );
}

export default PremiumContent;
