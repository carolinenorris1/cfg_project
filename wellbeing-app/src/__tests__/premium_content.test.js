import React from "react";
import renderer from "react-test-renderer";
import { CountryProvider } from "../context/CountryContext";

import PremiumContent from "../carousel/PremiumContent/PremiumContent";

it("renders correctly when user views the premium content with default country value on the carousel", () => {
  const tree = renderer
    .create(
      <CountryProvider>
        <PremiumContent />
      </CountryProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
