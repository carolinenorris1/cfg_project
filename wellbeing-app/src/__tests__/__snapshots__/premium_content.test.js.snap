// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when user views the premium content with default country value on the carousel 1`] = `
<div>
  <p>
    Want to discover even more great features?
  </p>
  <p>
    For just 
    <b>
      Â£2.99/month
    </b>
    , you can unlock even more tools to help you 
    <b>
      UnHook
    </b>
    .
  </p>
</div>
`;
